activerecord/test/cases/calculations_test.rb

```diff
@@ -58,7 +58,7 @@ def test_should_return_integer_average_if_db_returns_such
   end
 
   def test_should_return_nil_as_average
-    assert_nil NumericData.average(:bank_balance)
```

Comment:
This assertion actually would be failed without Ruby 2.6.0 if bigdecimal 1.4.0.pre.20181130a is used.

https://github.com/rails/rails/blob/e7717ec7a09ca83f92a9819303f7a93e038a2679/activerecord/lib/active_record/relation/calculations.rb#L400-L405

How about checking `nil.respond_to?(:to_d)`?

```diff
--- a/activerecord/test/cases/calculations_test.rb
+++ b/activerecord/test/cases/calculations_test.rb
@@ -57,8 +57,12 @@ def test_should_return_integer_average_if_db_returns_such
     assert_equal 3, value
   end
 
-  def test_should_return_nil_as_average
-    assert_nil NumericData.average(:bank_balance)
+  def test_should_return_nil_to_d_as_average
+    if nil.respond_to?(:to_d)
+      assert_equal BigDecimal(0), NumericData.average(:bank_balance)
+    else
+      assert_nil NumericData.average(:bank_balance)
+    end
   end
 
   def test_should_get_maximum_of_field
```

Reply:
Makes sense, Thanks for helping out 🙌